packages/protocol/test/ERC1724/ZkAssetOwnable.js

```diff
@@ -358,155 +400,147 @@ contract.skip('ZkAssetOwnable', (accounts) => {
+                depositInputOwnerAccounts,
+                transferInputNotes,
+                transferOutputNotes,
+            } = await setupTwoProofTest(noteValues);
+
+            const depositProof = new JoinSplitProof(
+                depositInputNotes,
+                depositOutputNotes,
+                sender,
+                depositPublicValue,
+                publicOwner,
+            );
+            const depositData = depositProof.encodeABI(zkAssetOwnable.address);
+            const depositSignatures = depositProof.constructSignatures(zkAssetOwnable.address, depositInputOwnerAccounts);
+            await ace.publicApprove(zkAssetOwnable.address, depositProof.hash, depositPublicValue, {
                 from: accounts[0],
             });
-            await zkAssetOwnable.confidentialTransfer(depositProof.proofData, depositProof.signatures);
-
-            const transferProof = proof.joinSplit.encodeJoinSplitTransaction({
-                inputNotes: notes.slice(0, 2),
-                outputNotes: notes.slice(2, 4),
-                senderAddress: accounts[0],
-                inputNoteOwners: [],
-                publicOwner: accounts[1],
-                kPublic: -40,
-                validatorAddress: zkAssetOwnable.address,
-            });
-            const transferProofOutput = outputCoder.getProofOutput(transferProof.expectedOutput, 0);
-            const transferProofHash = outputCoder.hashProofOutput(transferProofOutput);
-            await ace.publicApprove(zkAssetOwnable.address, transferProofHash, 40, {
-                from: accounts[1],
+            await zkAssetOwnable.confidentialTransfer(depositData, depositSignatures, { from: accounts[0] });
+            const transferPublicValue = -40;
+            const transferProof = new JoinSplitProof(
+                transferInputNotes,
+                transferOutputNotes,
+                sender,
+                transferPublicValue,
+                publicOwner,
+            );
+            const transferData = transferProof.encodeABI(zkAssetOwnable.address);
+
+            await ace.publicApprove(zkAssetOwnable.address, transferProof.hash, transferPublicValue, {
+                from: accounts[0],
             });
 
             // No confidentialApprove() call
+            await zkAssetOwnableTest.callValidateProof(JOIN_SPLIT_PROOF, transferData);
 
-            const opts = {
-                from: accounts[1],
-            };
-            await ace.publicApprove(zkAssetOwnable.address, transferProofHash, 0, opts);
-
-            await zkAssetOwnableTest.callValidateProof(JOIN_SPLIT_PROOF, transferProof.proofData);
-
-            const formattedProofOutput = `0x${transferProofOutput.slice(0x40)}`;
             await truffleAssert.reverts(
-                zkAssetOwnableTest.callConfidentialTransferFrom(JOIN_SPLIT_PROOF, formattedProofOutput),
+                zkAssetOwnableTest.callConfidentialTransferFrom(JOIN_SPLIT_PROOF, transferProof.eth.output),
```

Comment:
Double-check that this is not a false positive. Now there's both `proof.eth.output` and `proof.eth.outputs`.

Reply:
Done